### PR TITLE
fix(angular): use search client

### PR DESCRIPTION
**Pull request by francoischalifour**
## Description

This updates all the Angular examples to use `searchClient` instead of `appId` and `apiKey`.

## Related

- algolia/angular-instantsearch#496

### CODE REVIEW

File: Angular InstantSearch/debounced-search-box/package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "karma-jasmine-html-reporter": "0.2.2",
     "protractor": "6.0.0",
     "ts-node": "8.1.0",
-    "tslint": "5.16.0"
+    "tslint": "5.16.0",
+    "typescript": "2.7.2"
```
Comment (Haroenv) [inline Angular InstantSearch/debounced-search-box/package.json]: why such a low version of typescript?

File: Angular InstantSearch/geo-search/package.json [hunk 2]
```diff
@@ -33,7 +33,6 @@
     "@angular/cli": "7.3.8",
     "@angular/compiler-cli": "6.1.10",
     "@angular/language-service": "6.1.10",
-    "@types/algoliasearch": "3.27.8",
```
Comment (Haroenv) [inline Angular InstantSearch/geo-search/package.json]: why was this removed?

File: Angular InstantSearch/getting-started/src/app/app.component.ts [hunk 3]
```diff
@@ -1,4 +1,10 @@
 import { Component } from '@angular/core';
+import algoliasearch from 'algoliasearch/lite';
```
Comment (Haroenv) [inline Angular InstantSearch/getting-started/src/app/app.component.ts]: inconsistent prettier formatting here, the other ones don't set single quote

File: Angular InstantSearch/manipulating-lists/filtering/src/app/app.component.ts [hunk 4]
```diff
@@ -1,5 +1,11 @@
 import { Component } from '@angular/core';
 import { sortBy } from 'lodash';
+import algoliasearch from 'algoliasearch/lite';
```
Comment (Haroenv) [inline Angular InstantSearch/manipulating-lists/filtering/src/app/app.component.ts]: same comment about prettier config

File: Angular InstantSearch/manipulating-lists/sorting/src/app/app.component.ts [hunk 5]
```diff
@@ -1,5 +1,11 @@
 import { Component } from '@angular/core';
 import { sortBy } from 'lodash';
+import algoliasearch from 'algoliasearch/lite';
```
Comment (Haroenv) [inline Angular InstantSearch/manipulating-lists/sorting/src/app/app.component.ts]: quotes

File: Angular InstantSearch/debounced-search-box/package.json (hunk 1 above)
Comment (francoischalifour) [reply Angular InstantSearch/debounced-search-box/package.json]: I didn't change that. Yarn reordered the deps.

File: Angular InstantSearch/getting-started/src/app/app.component.ts (hunk 3 above)
Comment (francoischalifour) [reply Angular InstantSearch/getting-started/src/app/app.component.ts]: It is consistent with the file. We need to run Prettier on another PR.

File: Angular InstantSearch/manipulating-lists/filtering/src/app/app.component.ts (hunk 4 above)
Comment (francoischalifour) [reply Angular InstantSearch/manipulating-lists/filtering/src/app/app.component.ts]: It is consistent with the file. We need to run Prettier on another PR.

File: Angular InstantSearch/manipulating-lists/sorting/src/app/app.component.ts (hunk 5 above)
Comment (francoischalifour) [reply Angular InstantSearch/manipulating-lists/sorting/src/app/app.component.ts]: It is consistent with the file. We need to run Prettier on another PR.

File: Angular InstantSearch/geo-search/package.json (hunk 2 above)
Comment (francoischalifour) [reply Angular InstantSearch/geo-search/package.json]: Because we import `algoliasearch/lite` which does not have the types. So it's unused.

Comment (Haroenv) [reply Angular InstantSearch/geo-search/package.json]: lite does have the types though: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/algoliasearch/lite/index.d.ts

Comment (francoischalifour) [reply Angular InstantSearch/geo-search/package.json]: Oh interesting! So I have this error in the latest version:

> ERROR in node_modules/@types/algoliasearch/lite/index.d.ts(14,43): error TS2304: Cannot find name 'Exclude'.

This is the error raised in algolia/algoliasearch-client-javascript#753.

We can move forward with this PR and add them when fixed.

Comment (Haroenv) [reply Angular InstantSearch/geo-search/package.json]: As discussed, this is because the example uses 2.7, instead of a more recent version (2.8+) which comes with Exclude, which is written in the preamble of the definitions

**Comment by francoischalifour** [comment]
Moving forward with this PR.

We can reiterate on the Angular examples by upgrading TypeScript to ≥ 2.8 and adding the development dependency `@types/algoliasearch`. See https://github.com/algolia/algoliasearch-client-javascript/issues/753#issuecomment-487562574 for context.
